File: neo-physis-equa-backend/src/main/java/com/neo_physis_equa_backend/service/AuthService.java
package com.neo_physis_equa_backend.service;

import com.neo_physis_equa_backend.dto.AuthResponse;
import com.neo_physis_equa_backend.dto.LoginRequest;
import com.neo_physis_equa_backend.dto.RegistroRequest;
import com.neo_physis_equa_backend.entity.Usuario;
import com.neo_physis_equa_backend.exception.EmailAlreadyExistsException;
import com.neo_physis_equa_backend.exception.InvalidCredentialsException;
import com.neo_physis_equa_backend.repository.UsuarioRepository;
import com.neo_physis_equa_backend.security.JwtService;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
public class AuthService {

    private final UsuarioRepository usuarioRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtService jwtService;

    public AuthService(UsuarioRepository usuarioRepository, PasswordEncoder passwordEncoder, JwtService jwtService) {
        this.usuarioRepository = usuarioRepository;
        this.passwordEncoder = passwordEncoder;
        this.jwtService = jwtService;
    }

    public AuthResponse registro(RegistroRequest request) {
        if (usuarioRepository.existsByEmail(request.getEmail())) {
            throw new EmailAlreadyExistsException("El email ya está registrado");
        }

        Usuario usuario = new Usuario();
        usuario.setNombre(request.getNombre());
        usuario.setEmail(request.getEmail());
        usuario.setPassword(passwordEncoder.encode(request.getPassword()));

        usuarioRepository.save(usuario);

        String token = jwtService.generateToken(usuario);
        return new AuthResponse(token, usuario.getId(), usuario.getNombre(), usuario.getEmail(), "Registro exitoso");
    }

    public AuthResponse login(LoginRequest request) {
        Usuario usuario = usuarioRepository.findByEmail(request.getEmail())
                .orElseThrow(() -> new InvalidCredentialsException("Credenciales inválidas"));

        if (!passwordEncoder.matches(request.getPassword(), usuario.getPassword())) {
            throw new InvalidCredentialsException("Credenciales inválidas");
        }

        String token = jwtService.generateToken(usuario);
        return new AuthResponse(token, usuario.getId(), usuario.getNombre(), usuario.getEmail(), "Login exitoso");
    }

    public AuthResponse obtenerUsuarioPorEmail(String email) {
        Usuario usuario = usuarioRepository.findByEmail(email)
                .orElseThrow(() -> new RuntimeException("Usuario no encontrado"));
        return new AuthResponse(null, usuario.getId(), usuario.getNombre(), usuario.getEmail(), "Usuario autenticado");
    }
}

File: neo-physis-equa-backend/src/auth/auth.service.ts
import { ConflictException, Injectable, UnauthorizedException } from '@nestjs/common';
import { JwtService } from '@nestjs/jwt';
import * as bcrypt from 'bcryptjs';
import { UsersService } from '../users/users.service';
import { AuthResponse } from './dto/auth-response.dto';
import { LoginDto } from './dto/login.dto';
import { RegisterDto } from './dto/register.dto';

@Injectable()
export class AuthService {
  constructor(
    private readonly usersService: UsersService,
    private readonly jwtService: JwtService,
  ) {}

  async registro(dto: RegisterDto): Promise<AuthResponse> {
    const existing = await this.usersService.findByEmail(dto.email);
    if (existing) {
      throw new ConflictException('El email ya está registrado');
    }

    const user = await this.usersService.create(dto);
    const token = this.jwtService.sign({ sub: user.id, email: user.email, role: user.role });
    return {
      token,
      id: user.id,
      nombre: user.name,
      email: user.email,
      mensaje: 'Registro exitoso',
      role: user.role,
      accessibilityProfile: user.accessibilityProfile,
      disabilityType: user.disabilityType,
    };
  }

  async login(dto: LoginDto): Promise<AuthResponse> {
    const user = await this.usersService.findByEmail(dto.email);
    if (!user) {
      throw new UnauthorizedException('Credenciales inválidas');
    }

    const passwordValid = await bcrypt.compare(dto.password, user.password);
    if (!passwordValid) {
      throw new UnauthorizedException('Credenciales inválidas');
    }

    const token = this.jwtService.sign({ sub: user.id, email: user.email, role: user.role });
    return {
      token,
      id: user.id,
      nombre: user.name,
      email: user.email,
      mensaje: 'Login exitoso',
      role: user.role,
      accessibilityProfile: user.accessibilityProfile,
      disabilityType: user.disabilityType,
    };
  }

  async perfil(id: string): Promise<AuthResponse> {
    const user = await this.usersService.findById(id);
    if (!user) {
      throw new UnauthorizedException('Usuario no encontrado');
    }
    return {
      token: null,
      id: user.id,
      nombre: user.name,
      email: user.email,
      mensaje: 'Usuario autenticado',
      role: user.role,
      accessibilityProfile: user.accessibilityProfile,
      disabilityType: user.disabilityType,
    };
  }
}
